atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -246,43 +255,84 @@ protected void initializeFromFreshInstance(List<String> addrList, int replicatio
+    }
+
     private void upgradeFromOlderInternalSchema(Client client) throws NotFoundException, InvalidRequestException, TException {
         Map<String, byte[]> metadataForTables = getMetadataForTables();
         Map<String, byte[]> tablesToUpgrade = Maps.newHashMapWithExpectedSize(metadataForTables.size());
+        String keyspace = configManager.getConfig().keyspace();
 
-        final CassandraKeyValueServiceConfig config = configManager.getConfig();
-        String keyspace = config.keyspace();
-        boolean safetyDisabled = config.safetyDisabled();
         for (CfDef clusterSideCf : client.describe_keyspace(keyspace).getCf_defs()) {
-            String tableName = fromInternalTableName(clusterSideCf.getName());
+            String tableName = internalTableName(clusterSideCf.getName());
             if (metadataForTables.containsKey(tableName)) {
                 byte[] clusterSideMetadata = metadataForTables.get(tableName);
-                CfDef clientSideCf = getCfForTable(tableName, clusterSideMetadata);
+                CfDef clientSideCf = getCfForTable(clusterSideCf.getName(), clusterSideMetadata);
                 if (!CassandraKeyValueServices.isMatchingCf(clientSideCf, clusterSideCf)) { // mismatch; we have changed how we generate schema since we last persisted
                     log.warn("Upgrading table {} to new internal Cassandra schema", tableName);
                     tablesToUpgrade.put(tableName, clusterSideMetadata);
                 }
-            } else if (!HIDDEN_TABLES.contains(tableName)) { // only expected cases
-                // Possible to get here from a race condition with another server starting up and performing schema upgrades concurrent with us doing this check
-                throw new RuntimeException(new UpgradeFailedException("Found a table " + tableName + " that did not have persisted AtlasDB metadata. If you recently did a Palantir update, try waiting until schema upgrades are completed on all backend servers and restarting this service."));
+            } else if (!tableName.equals(CassandraConstants.METADATA_TABLE)) { // only expected case
+                // Possible to get here from a race condition with another dispatch starting up and performing schema upgrades concurrent with us doing this check
+                log.error("Found a table " + tableName + " that did not have persisted Atlas metadata."
+                        + "If you recently did a Palantir update, try waiting until schema upgrades are completed on all backend CLIs/dispatches etc and restarting this service."
+                        + "If this error re-occurs on subsequent attempted startups, please contact Palantir support.");
             }
         }
 
-        // we are racing another server to do these same operations here, but they are idempotent / safe
+        // we are racing another dispatch to do these same operations here, but they are idempotent / safe
```

Comment:
you're overwriting a change here that carrino did on purpose (de-PGizing the comments)